### ISSUE TITLE
Re-consider a more user-friendly anchor symbol, or make the default FALSE for anchor_sections

### ISSUE THREAD

**Issue by apreshill** [rstudio/rmarkdown#1946]
Re: https://github.com/rstudio/rmarkdown/pull/1884

I think the default `anchor_sections = TRUE` is hard as a default given the `#` symbol. It isn’t intuitive or self-explanatory, and most people don’t know that those are called anchors, so finding the argument `anchor_sections` to turn it off is hard to figure out. Also, because the hashtag symbol has a special meaning as a header in markdown syntax, it looks like a weird markdown rendering error when you see it- I even knew this feature was in place and STILL had a freak out and initially thought I was seeing an odd markdown parsing artifact at first.

On the other hand, precisely because many users may not be familiar with the concept of anchor headers (even if they enjoy them!), it will be hard to figure out that you can turn it _ON_ as well if the default is `FALSE`. And I think the utility is extremely useful for users (adding anchor symbols to my HTML-based R Markdown outputs is often the first thing I do, and the first thing I asked my intern Desiree to do!- thank you @atusy for recognizing that need and adding the functionality 🌟 ).

But, the more I use it in my own recent work across output formats and see users attempting to trouble-shoot this new feature, I think that if we want to keep `TRUE` as the default, we should switch to an svg symbol. This is familiar to many users because it is how GitHub `README`s look, `pkgdown` websites, and now `distill` as well, and there is even a link emoji :link:. All of the above boils down to:

+ If we keep the `#` symbol, I lean towards setting `FALSE` as the default across the board.
+ If we are open to an svg (single file, open source like the link icon here: https://material.io/resources/icons/?search=link&style=baseline), I think `TRUE` can stay as the default (with slide decks and perhaps pagedown setting it to `FALSE`)

## Some examples with svg icons

<img width="943" alt="Screen Shot 2020-11-10 at 1 21 54 PM" src="https://user-images.githubusercontent.com/12160301/98715124-b8884f80-2357-11eb-9de6-1a755eafe881.png">

<img width="834" alt="Screen Shot 2020-11-10 at 12 01 47 PM" src="https://user-images.githubusercontent.com/12160301/98714758-49126000-2357-11eb-86d3-eabbe7c29e14.png">
<img width="834" alt="Screen Shot 2020-11-10 at 12 02 18 PM" src="https://user-images.githubusercontent.com/12160301/98714771-4c0d5080-2357-11eb-9d1d-20794d2e8c86.png">
<img width="834" alt="Screen Shot 2020-11-10 at 12 02 55 PM" src="https://user-images.githubusercontent.com/12160301/98714778-4e6faa80-2357-11eb-820f-285dc3658d07.png">
<img width="834" alt="Screen Shot 2020-11-10 at 12 03 24 PM" src="https://user-images.githubusercontent.com/12160301/98714783-516a9b00-2357-11eb-9af5-165ba070c524.png">
<img width="834" alt="Screen Shot 2020-11-10 at 1 19 34 PM" src="https://user-images.githubusercontent.com/12160301/98714857-65160180-2357-11eb-9cac-ba30f4ff549c.png">

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by apreshill** [comment]
I forgot there is also this unicode solution. I prefer the aesthetics of the simpler svg icon, but for thoroughness: 

https://discourse.gohugo.io/t/adding-anchor-next-to-headers/1726/14

Can be seen used on tidymodels.org:
<img width="943" alt="Screen Shot 2020-11-10 at 1 42 43 PM" src="https://user-images.githubusercontent.com/12160301/98717114-a5c34a00-235a-11eb-8527-bc889abd6965.png">

Currently implemented as a Hugo partial, before the Hugo render hooks were implemented:
https://github.com/tidymodels/tidymodels.org/blob/master/layouts/partials/hanchor.html

**Comment by yihui** [comment]
I have already agreed and suggested that we make `anchor_sections = FALSE` by default two weeks ago: https://github.com/rstudio/pagedown/issues/194#issuecomment-718828351 So @cderv please go ahead and just do that :)

**Comment by yihui** [comment]
I forgot to mention that we could also mention an example like 

```css
a.anchor-section::before {
    content: '🔗';
}
```

so people who have strong opinions on the symbol know how to use their favorite symbol. I'm not sure where this example should live, though.

**Comment by cderv** [comment]
So what I’ll do is 

* revert to FALSE as default but not change the hash symbol for TRUE
* but make it clearer and easier to customize the content. At least documentation - I’ll find a place.

**Comment by dpprdan** [comment]
FWIW

```css
a.anchor-section::before {content: '\01F517\00FE0E';}
```

or

```css
a.anchor-section::before {content: '🔗\00FE0E';}
```

(`\00FE0E` is the 'text-style' unicode variation selector [in CSS](https://graphemica.com/%EF%B8%8E). Without it, the emoji will (usually) be shown AFAIU).

With this I get

![grafik](https://user-images.githubusercontent.com/1423562/98835877-45f39e80-2441-11eb-80a2-b7425647d802.png)

with the standard R Markdown template in Rstudio on Windows 10.

But the exact symbol will probably depend on the OS/browser/font combination, AFAIK. Because Alison's tidymodels example above looks like this for me:

![grafik](https://user-images.githubusercontent.com/1423562/98836394-ecd83a80-2441-11eb-956e-2f679266d87b.png)

**Comment by yihui** [comment]
@cderv Sounds good to me! Thanks!

@dpprdan Oh these are awesome tips! Thank you so much!

**Comment by jtr13** [comment]
I realize this has already been decided, but for the record, I love the new `anchor_sections` feature, so sad to see it not be the default (agree with @apreshill that a link would be a better symbol choice). I have a different issue which is that the links don't work consistently. Hard to reproduce behavior that fails *sometimes*, but here's one example that doesn't always work for me: https://edav.info/github.html#now-what. I thought it might be an **rmarkdown** version issue but since there's only one version of **rmarkdown** (2.5) that produces the anchor links, that's not it. Maybe a browser issue? It's odd since clicking on the subheading in the toc always works. 🤷‍♀️

**Comment by cderv** [comment]
Thanks for your feedback @jtr13 ! 

We thought it was good to have this in all document! But having `anchor_sections` has default in `html_document` is not without consequences because (lots of) other formats are based on this, and they may not handle well the anchor addition. Also for very short document produced with `html_document`, you may not want anchors being added. At the end, adding a new feature and activating it by default in `html_document` can change behavior for all past and present document. That is mainly why we will set it to `FALSE by default in `html_document` now. 

But for other format more focused, like `bookdown::gitbook()`, we may default to TRUE. In this long form document with a more end-usage oriented formats, it could make more sense to have it to TRUE. Anyway, it is really easy to activate it so the default is not so important.

As the bonus, reverting to FALSE the default could be accompanyied by the possibility to style the anchor. This is currently in dev in the associated PR. 

About you issue, I can't tell for sure where it could come from. If you encounter it reliably and/or can reproduce, we could look into it. anchor links are using the id of the associated header. If you did not provide one, a default one will be generated (by pandoc) based on the header content. If you change the header content, the id will change and hence your link will no more work. To prevent that, setting an id manually for each of your header is the best option. But obviously what I am saying will occur only if you change the headers of your document. Anyway, if you have more info on this, please open a new issue!

**Comment by jtr13** [comment]
All makes sense... thanks so much for the informative explanation!

**Comment by yihui** [comment]
@jtr13 The gitbook layout and CSS are fairly complicated, and I'm not surprised that the anchors don't work sometimes. I've been thinking of redesigning it for long, but still haven't found a chance. Anyway, I'm totally for `anchor_sections = TRUE` by default in `bookdown::gitbook()`, and I just made that change in the dev version of **bookdown**. Thanks!

**Comment by jtr13** [comment]
Thank you!! ⚓ ⚓ ⚓

**Comment by apreshill** [comment]
FWIW @jtr13 I am in firefox and the anchor section you linked to works for me :)

**Comment by jtr13** [comment]
Does it work if you refresh the page? I've been getting very weird behavior: it works the first time but not on refresh. Or sometimes every other time. ???

**Comment by apreshill** [comment]
Works every time for me 🤷‍♀️

**Comment by cderv** [comment]
FWIW we closed this issue as we have changed the default to FALSE. But there was a OR in this issue, and so there is still some more coming up regarding user-friendly configuration about the anchor section customization. PR #1964 is still opened - we may just do a needed patch release on CRAN for **rmarkdown** and includes the improvements in a following version, after having time to gather some feedback!
